### ISSUE TITLE
500: Internal Server Error after login (new install)

### ISSUE THREAD

**Issue by srdco** [kimai/kimai#342]
**Describe the bug**
As soon as I login (brand new install), it give a 500 Server Error.

**To Reproduce**
1. Install Kimai2 on localhost (Ubuntu 18.04 LTS with Apache 2.4+) using standard instructions (replace wwwdata with standard user) with MySQL database. Installed under /var/www/kimai2.
2. Setup Virtual Host in /etc/apache2/sites-enabled/kimai.conf/ (for kimai.local) and add entry in /etc/hosts mapping 127.0.0.1 to kimai.local
3. Try to login.

**Expected behavior**
Display timesheet / dashboard? (It's trying to reach 

**Screenshots**
Blank screen. Header shows it was redirected from kimai.local (302) --> /en (301) -> /en/timesheet/ (302 --> /en/login (500)

Output from kimai-access.log: 
127.0.0.1 - - [05/Oct/2018:23:09:38 -0500] "GET / HTTP/1.1" 302 531 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:61.0) Gecko/20100101 Firefox/61.0"
127.0.0.1 - - [05/Oct/2018:23:09:39 -0500] "GET /en/timesheet/ HTTP/1.1" 302 663 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:61.0) Gecko/20100101 Firefox$
127.0.0.1 - - [05/Oct/2018:23:09:39 -0500] "GET /en/login HTTP/1.1" 500 185 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:61.0) Gecko/20100101 Firefox/61.0"

Error log is showing nothing.

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04 LTS
 - Browser: Firefox 
 - Version: 61

**Additional context**
Successfully used Kimai v1 on a previous installation of Ubuntu on this computer.
Desktop dual workstation and local/self server. Successfully using PHPMyAdmin and a PHPIP (PHP/MySQL internal website) installation on it.
Already tried, completely removed Kimai2, and installed fresh again. Tried editing htaccess (removing the 2 'optional' lines).
During installation, had to substitute sre (standard username) for wwwdata during installation to overcome permission errors in the composer command. However, went back and re-CHOWN / CHMOD the folder per the instructions.

Any pointers would be appreciated!

**Comment by kevinpapst** [comment]
Hi @srdco!
Lets try to find out ... if this is a local installation, you could first check if its a permission problem, apply 777 recursive on `var/`. Then check the Kimai log file at `var/log/`.
Also add a CustomErrorLog to the Apache Host and see if something shows up in there.

**Comment by srdco** [comment]
Ran `sudo chmod -R 777 var/`. No change.
Added `LogLevel debug` to apache2 configuration file. NOTE: Already had config file setup for logs like so: 
> ErrorLog /var/log/apache2/kimai-error.log
> CustomLog /var/log/apache2/kimai-access.log combined

Result in the kimai-error.log after changing LogLevel and trying to access kimai.local once (it's already 'logged in', so it immediately redirects to http://kimai.local/en/timesheet/):

> [Sat Oct 06 14:50:14.582216 2018] [authz_core:debug] [pid 7483] mod_authz_core.c(809): [client 127.0.0.1:43370] AH01626: authorization result of Require all granted: granted
> [Sat Oct 06 14:50:14.582269 2018] [authz_core:debug] [pid 7483] mod_authz_core.c(809): [client 127.0.0.1:43370] AH01626: authorization result of <RequireAny>: granted
> [Sat Oct 06 14:50:14.582377 2018] [authz_core:debug] [pid 7483] mod_authz_core.c(809): [client 127.0.0.1:43370] AH01626: authorization result of Require all granted: granted
> [Sat Oct 06 14:50:14.582390 2018] [authz_core:debug] [pid 7483] mod_authz_core.c(809): [client 127.0.0.1:43370] AH01626: authorization result of <RequireAny>: granted

**Comment by kevinpapst** [comment]
I have no Apache setup, so I can test it right now. Have you seen any page by now?
Or does the first call already fail? If the redirects happen, the general setup seems to work, but no page can be displayed.
Are the Kimai logfiles at `var/log/` (in the Kimai directory) empty?
Try to run `bin/console cache:clear` and open the page WITHOUT calling `bin/console cache:warmup`! The cache will be warmed up on-the.fly and a log message should be dumped.

**Comment by srdco** [comment]
Just tried running `bin/console cache:clear`. It said it was successfully cleared with debug=false.
Still same result when accessing kimai.local (500-server-error, blank page).
prod.log showed difficulties creating files/folders in var/cache, so I ran `sudo chown -R www-data var/` in the kimai2 folder, and tried to reaccess kimai.local. No longer blank, but a written 500-server-error page.
Prod.log errors changed, and I don't quite follow them. New prod.
[prod.log](https://github.com/kevinpapst/kimai2/files/2453382/prod.log)
log file attached.

**Comment by kevinpapst** [comment]
Thats the important part:
```
[2018-10-06 15:25:15] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\ClassNotFoundException: "Attempted to load class "IntlTimeZone" from the global namespace. Did you forget a "use" statement?" at /var/www/kimai2/vendor/twig/extensions/lib/Twig/Extensions/Extension/Intl.php line 71 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\ClassNotFoundException(code: 0): Attempted to load class \"IntlTimeZone\" from the global namespace.\nDid you forget a \"use\" statement? at /var/www/kimai2/vendor/twig/extensions/lib/Twig/Extensions/Extension/Intl.php:71)"} []
```
You are missing the INTL extension:
```
sudo apt-cache search php|grep intl
```
So try something like 
```
sudo apt-get install php7.2-intl
sudo /etc/init.d/apache2 restart
```
I'll check why you didn't get any feedback during the composer installation. You should be warned that there are missing dependencies.

**Comment by kevinpapst** [comment]
Hm weird, its here: https://github.com/kevinpapst/kimai2/blob/master/composer.json#L8
You didn't get any warning when executing `composer install`? Did you install with the `--ignore-platform-reqs` flag?

**Comment by srdco** [comment]
That's strange.
I installed the php7.2-intl extension prior to the first install. I went ahead and ran the command again, just to be sure, but it shows installed. But info.php doesn't show it. Let me see if I can figure out why it wasn't activated.
If it's installed but not activated, that may be why composer didn't flag it. No I didn't receive warnings during the install, and no I didn't use --ignore-platform-reqs flag.

**Comment by kevinpapst** [comment]
Check with `php -m` if its activated. 
I am not using Ubuntu, but try `sudo phpenmod intl`. Restart Apache afterwards.
If `phpenmod` doesn't exist, you probably need to symlink the module into the correct folder.  For ngnivnx this is the folder `/etc/php/7.1/fpm/conf.d/`, but I don't know how to do it for Apache.

**Comment by srdco** [comment]
Solved. PHP7.2 is (of course) latest stable release, and the 'default' one for Ubuntu 18.04. For some reason, it was still using PHP7.1 ... at least partially? So ... I switched from PHP7.1 to 7.2, and it appears to be working as expected. 
For reference of anyone else:
`sudo a2dismod php7.1
sudo a2enmod php7.2
systemctl restart apache2`

Thanks so much for helping debug this! Looking forward to using Kimai2 - thanks for all your work!
